### PR TITLE
Review Docker management

### PR DESCRIPTION
**Pull request by Odyno**
My fix linked with the Issue #934 

- Renew Dockerfile
- Add common script for docker management ( create base image with complete enviroment, run bash in complete enviroment, run build script on complete enviroment )

### CODE REVIEW

**Comment by homeassistant** [comment]
Hi @Odyno, 

It seems you haven't yet signed a CLA. Please do so [here](https://home-assistant.io/developers/cla_sign_start/?pr=home-assistant&#x2F;home-assistant-polymer#947).

Once you do that we will be able to review and accept this pull request.

Thanks!

File: Dockerfile [hunk 1]
```diff
@@ -1,25 +1,41 @@
-FROM node:8.2.1-alpine
+FROM node:8.9
```
Comment (balloob) [inline Dockerfile]: All of Home Assistant infrastructure runs on Alpine and it runs great. I don't see any reason to change that.

File: Dockerfile [hunk 2]
```diff
@@ -1,25 +1,41 @@
...
+
+RUN sed -i -e 's/# en_US.UTF-8 UTF-8/en_US.UTF-8 UTF-8/' /etc/locale.gen && \
+  dpkg-reconfigure --frontend=noninteractive locales && \
+  update-locale LANG=en_US.UTF-8
+
+ENV LANG en_US.UTF-8
+
+RUN curl -o- -L https://yarnpkg.com/install.sh | bash
+
+RUN update-alternatives --install /usr/bin/python python /usr/bin/python2.7 1 \
```
Comment (balloob) [inline Dockerfile]: Neither of these Python versions are supported by Home Assistant 🤔

File: package-lock.json [hunk 3]
```diff
@@ -0,0 +1,33 @@
+{
```
Comment (balloob) [inline package-lock.json]: Please do not commit this file. We use yarn.

File: Dockerfile (hunk 1 above)
Comment (Odyno) [reply Dockerfile]: ... and my question is, **why not?** 

I checked home-assistant and it start from ["python:3.6"](https://github.com/home-assistant/home-assistant/blob/dev/Dockerfile#L5) that start from ["buildpack-deps:jessie"](https://github.com/docker-library/python/blob/a1aa406bfd8c7b129e6e0ee0ba972b863624ac0d/3.6/jessie/Dockerfile#L7) , so why use alpine on home-assistant-polymer?

The concept behind Alpine was to create a light and simple firewall/proxy distro that could run in RAM... it's very useful for "production docker" but has less sense in a development environment.
This environment docker serves only to develop and create the frontend package, it not provide service for the production. This is so true that the first step that you do is to [add/install the build and versioning files](https://github.com/home-assistant/home-assistant-polymer/blob/master/Dockerfile#L7)

Lastly, I have shown that it does not work ( Issue #934 ) and I have not found any automatic system that validates the current compilation via docker (if that is not the case please tell me so. I have searched on travis but no dockers are used, not even I have not seen web-hooks pointing on docker hub)

**Why I choose node:8.9 and not node:8.2.1?**
The home-assistant-polymer (as user of node), why should it be frozen even at patch version? the idea is to use official LTS provided by NodeJS and do not loose all the security and optimization patches ( and finally it start from "buildpack-deps:jessie" the same of "python:3.6" on "home-assistant" )

File: Dockerfile (hunk 2 above)
Comment (Odyno) [reply Dockerfile]: I checked the old docker, and noticed this:
```
Last login: Wed Feb 27 23:21:35 on ttys000
astaniscia@spa-079-l:~$ docker run -it node:8.2.1-alpine sh
/ # apk update && apk add python3
fetch http://dl-cdn.alpinelinux.org/alpine/v3.6/main/x86_64/APKINDEX.tar.gz
fetch http://dl-cdn.alpinelinux.org/alpine/v3.6/community/x86_64/APKINDEX.tar.gz
v3.6.2-271-g168bada463 [http://dl-cdn.alpinelinux.org/alpine/v3.6/main]
v3.6.2-266-g0f1867409e [http://dl-cdn.alpinelinux.org/alpine/v3.6/community]
OK: 8443 distinct packages available
(1/11) Installing libbz2 (1.0.6-r5)
(2/11) Installing expat (2.2.0-r1)
(3/11) Installing libffi (3.2.1-r3)
(4/11) Installing gdbm (1.12-r0)
(5/11) Installing xz-libs (5.2.3-r0)
(6/11) Installing ncurses-terminfo-base (6.0_p20171125-r0)
(7/11) Installing ncurses-terminfo (6.0_p20171125-r0)
(8/11) Installing ncurses-libs (6.0_p20171125-r0)
(9/11) Installing readline (6.3.008-r5)
(10/11) Installing sqlite-libs (3.20.1-r1)
(11/11) Installing python3 (3.6.1-r3)
Executing busybox-1.26.2-r5.trigger
OK: 71 MiB in 24 packages
/ # python3 --version
Python 3.6.1
/ #
```

can you confirm  this version? **Python 3.6.1**

File: package-lock.json (hunk 3 above)
Comment (Odyno) [reply package-lock.json]: OK... it will be added on the git ignore

File: Dockerfile (hunk 1 above)
Comment (mcspr) [reply Dockerfile]: @Odyno You can still use `-alpine` though https://github.com/nodejs/docker-node/blob/8498ff5302c19506c9edc3ce152f2bd4aa6b26b3/8/alpine/Dockerfile
`gcc, make` etc. could be easily pruned. and python3 being up-to-date looks not so relevant here.

Other changes would be nice to have, easily get compiled frontend using just docker.

**Comment by Odyno** [comment]
Closed in favour of PR #1113

**Pull request closed without merging**